dev_docs/07_visual_testing_guide.md

```diff
@@ -0,0 +1,101 @@
+
+     ```bash
+     yarn install
+     ```
+
+   - Set the `PERCY_TOKEN` environment variable:
+
+     ```bash
+     export PERCY_TOKEN=<your-percy-token>
+     ```
+
+2. **Run Visual Tests**:
+   - Execute the tests with the visual testing configuration:
+
+     ```bash
+     yarn test:visual
+     ```
+
+3. **Check results**:
+   - Head over to the Percy Dashboard and check the results for the latest Percy build.
+
+### Visual testing workflow
+
+  We use GitHub Actions to execute the visual testing workflow on every PR. When changes to a PR are made, a deployment request is initiated. Only **Learning Equality team members** can approve this request. Once approved, the deployment is executed, and visual tests are run automatically. The results of the test run are surfaced in the form of an automated comment containing a link to the Percy build.
+
+  **Note:** Developers outside of Learning Equality can only review the visual changes for local test runs. To review the visual tests executions that run on GitHub Actions, one needs to be a part of Learning Equality's Browserstack team.
+
+### Writing visual tests
+
+  You can write the visual tests alongside the unit tests, i.e. in the same test file. Take a look at [`KButton.spec.js`](../lib/buttons-and-links/__tests__/KButton.spec.js) to familiarize yourself with writing visual tests.
+
+1. **Import Utility Functions**:
+   - Import the utility functions from [`visual.testUtils.js`](../jest.conf/visual.testUtils.js):
+
+     ```javascript
+     import { renderComponent, takeSnapshot } from './visual.testUtils';
+     ```
+
+     You can also import and use the utility functions for managing component's visual states as needed. To get an overview of the different utility functions available, please check out [`visual.testUtils.js`](../jest.conf/visual.testUtils.js).
+
+2. **Write Tests**:
+   - Use the utility functions to render components and take snapshots. For example:
+
+     ```javascript
+     describe.visual('KButton Visual Tests', () => {
+       it('Sample test for KButton', async () => {
+         await renderComponent('KButton', { text: 'Raised Button', appearance: 'raised-button' });
+         await takeSnapshot('KButton - Raised Button', { widths: [375, 520] });
+       });
+     });
+     ```
+
+   - Make sure to use `describe.visual` or `it.visual` instead of the default notations for writing test blocks containing visual tests so as to prevent any unexpected behavior. These custom blocks add a [Visual] tag to the test name whose presence or absence are then checked using a regex pattern based on the type of tests executed. This implementation helps us to determine which test blocks should be executed by Jest and which ones should be skipped.
+
+3. **Simulate User Interactions**:
+   - Use the custom commands to simulate user interactions. For example, to simulate the *'click'* user event, you can do something like:
+
+     ```javascript
+      await click('button');
```

Comment:
I haven't found this helper in `visual.testUtils.js` and neither other helpers mentioned in the latter section. I am not familiar with the project - is it work in progress?

Reply:
Idem. This was added [here](https://github.com/learningequality/kolibri-design-system/pull/710/files#diff-8f2a39889353766cd3e42a4127bd043067c60dd97d054d2684622bbf0babc4dbR84) :sweat_smile:.